### PR TITLE
feat: filter based on insufficient lookahead

### PR DESCRIPTION
**Pull request by MartinBernstorff**
Automatically created PR from invoke

### CODE REVIEW

File: psycop/common/feature_generation/sequences/cohort_definer_to_prediction_times.py [hunk 1]
```diff
@@ -34,14 +34,23 @@ def _polars_dataframe_to_patient_timestamp_mapping(
         dataframe: pl.DataFrame,
         id_col_name: str,
         patient_timestamp_col_name: str,
+        lookahead: dt.timedelta | None = None,
     ) -> dict[PATIENT_ID, list[dt.datetime]]:
+        max_timestamp: dt.datetime = dataframe[patient_timestamp_col_name].max()  # type: ignore
```
Comment (MartinBernstorff) [inline psycop/common/feature_generation/sequences/cohort_definer_to_prediction_times.py:39]: Would be even better to get the max timestamp of (outcome_timestamps | prediction_time_timestamps | predictors), but this is much faster, and probably just as good.

File: psycop/common/feature_generation/sequences/cohort_definer_to_prediction_times.py [hunk 2]
```diff
@@ -34,14 +34,34 @@ def _polars_dataframe_to_patient_timestamp_mapping(
...
+        Args:
+            dataframe: Polars dataframe with patient ids and timestamps.
+            id_col_name: Name of the column with patient ids.
+            patient_timestamp_col_name: Name of the column with timestamps.
+            lookahead: If not None, only timestamps before the max timestamp minus the lookahead
+                will be included. E.g. for a lookahead of 2 years, this ensures we actually have 2
+                years of data to label the outcome.
+        """
+        if lookahead is not None:
+            max_timestamp: dt.datetime = dataframe[patient_timestamp_col_name].max()  # type: ignore
```
Comment (MartinBernstorff) [inline psycop/common/feature_generation/sequences/cohort_definer_to_prediction_times.py:50]: Would be even better to get the max timestamp of (outcome_timestamps | prediction_time_timestamps | predictors), but this is much faster, and probably just as good.

File: psycop/common/feature_generation/sequences/cohort_definer_to_prediction_times.py [hunk 3]
```diff
@@ -34,14 +34,34 @@ def _polars_dataframe_to_patient_timestamp_mapping(
...
+            dataframe = dataframe.filter(
+                pl.col(patient_timestamp_col_name) < max_timestamp - lookahead,
```
Comment (sarakolding) [inline psycop/common/feature_generation/sequences/cohort_definer_to_prediction_times.py:52]: <=?

**Review by sarakolding — APPROVED** [review]
beautiful

**Pull request merged**